src/Database/Driver/PDODriverTrait.php

```diff
@@ -94,9 +94,8 @@ public function isConnected()
                 $connected = false;
             }
         }
-        $this->connected = !empty($connected);
```

Comment:
Ah I understand. It seems to have already been dead when it was born.
https://github.com/cakephp/cakephp/pull/7690/files#diff-5accaf4e5f3bb02fed9c676b1d31c904R95